resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java

```diff
@@ -75,10 +76,7 @@ public DefaultDnsCache() {
     public DefaultDnsCache(int minTtl, int maxTtl, int negativeTtl) {
         this.minTtl = Math.min(Cache.MAX_SUPPORTED_TTL_SECS, checkPositiveOrZero(minTtl, "minTtl"));
         this.maxTtl = Math.min(Cache.MAX_SUPPORTED_TTL_SECS, checkPositiveOrZero(maxTtl, "maxTtl"));
-        if (minTtl > maxTtl) {
-            throw new IllegalArgumentException(
-                    "minTtl: " + minTtl + ", maxTtl: " + maxTtl + " (expected: 0 <= minTtl <= maxTtl)");
-        }
+        checkClosedInterval(minTtl, 0, maxTtl, "minTtl");
```

Comment:
This is going to give a misleading error message.

Ditto for DefaultDnsCnameCache.java